new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Pod.java

```diff
@@ -0,0 +1,150 @@
+import java.util.concurrent.Callable;
+
+import org.awaitility.core.ConditionFactory;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+public class Pod {
+
+  /**
+   * Check pods are restarted in the same order as in the pods list.
+   * @param pods names of the pods in a list
+   * @param namespace name of the namespace in which to check for pods rolling restart
+   * @return true if pods are restarted in rolling fashion
+   */
+  public static boolean verifyRollingRestartOccurred(ArrayList<String> pods, String namespace) {
+
+    // check the pods list is not empty
+    if (pods.isEmpty()) {
+      logger.severe("The pods list is empty");
+      return false;
+    }
+
+    // reusable condition factory
+    ConditionFactory retry
+        = with().pollInterval(5, SECONDS).atMost(5, MINUTES).await();
+
+    // check pods are terminated and started.
+    for (var pod : pods) {
+      retry
+          .conditionEvaluationListener(condition -> logger.info("Waiting for pod {0} to be "
+          + "terminating in namespace {1} "
+          + "(elapsed time {2}ms, remaining time {3}ms)",
+          pod,
+          namespace,
+          condition.getElapsedTimeInMS(),
+          condition.getRemainingTimeInMS()))
+          .until(assertDoesNotThrow(() -> onlyGivenPodTerminating(pods, pod, namespace),
+              String.format("pod %s didn't terminate in namespace %s", pod, namespace)));
+
+      retry
+          .conditionEvaluationListener(condition -> logger.info("Waiting for pod {0} to be ready in namespace {1} "
+          + "(elapsed time {2}ms, remaining time {3}ms)",
+          pod,
+          namespace,
+          condition.getElapsedTimeInMS(),
+          condition.getRemainingTimeInMS()))
+          .until(assertDoesNotThrow(() -> podReady(namespace, null, pod),
+              String.format("pod %s is not ready in namespace %s", pod, namespace)));
+    }
+
+    return true;
+  }
+
+  /**
+   * Return true if the given pod is the only one terminating from the list of pods.
+   *
+   * @param pods names of the pods in a list
+   * @param podName name of pod to check for termination status
+   * @param namespace name of the namespace in which the pod terminating status to be checked
+   * @return true if given pod is the only pod terminating
+   * @throws Exception when more than one pod is terminating or cluster query fails
+   */
+  public static Callable<Boolean> onlyGivenPodTerminating(ArrayList<String> pods, String podName, String namespace)
+      throws Exception {
+    return () -> {
+      int terminatingPods = 0;
+      boolean givenPodTerminating = false;
+      for (var pod : pods) {
+        if (Kubernetes.isPodTerminating(namespace, null, pod)) {
+          terminatingPods++;
```

Comment:
print the pod names that are terminating

Reply:
Its printed inside the Kubernetes.isPodTerminating method